driver-core/src/main/com/mongodb/ReadPreference.java

```diff
@@ -504,9 +505,7 @@ public static TaggableReadPreference nearest(final List<TagSet> tagSetList,
      * @return the read preference
      */
     public static ReadPreference valueOf(final String name) {
-        if (name == null) {
-            throw new IllegalArgumentException();
-        }
+        notNull("name", name);
```

Comment:
This avoids a warning for checking a parameter for null that is known to be non-null.  We need these checks in the public API for applications that are ignoring the nullability annotations.